### DISCUSSION TITLE
What's the criteria for a "confirmed" QSO?

### DISCUSSION THREAD

**Discussion by tabascoterrier** [magicbug/Cloudlog#1242] [Q&A] [answered]
I've recently set up a self hosted Cloudlog instance and I'm really enjoying it, my thanks to everyone involved for all your hard work!

Could someone tell me, what counts as a green "confirmed" QSO square on things like the gridsquare map?

I have eSQL configured and QSOs have a green uploaded and downloaded icon.  On the front page, my "Countries Breakdown" says "Confirmed 0 / 65 / 0", but all my gridsquares are red / worked but not confirmed (as are the table cells in things like the DXCC awards screen). So presumably there's some other criteria that I've yet to figure out.

73, M7SXB

**Comment by dg9vh** [comment]
Confirmed means we have 3 ways actually a "confirmation" can reach you: via
paper qsl (first number), via eQSL (second number) and via LoTW (third
number). A confirmation is then given, if a corresponding QSL (or eQSL or
LoTW) reaches you.

73 de Kim
DG9VH

Am Mo., 18. Okt. 2021 um 13:09 Uhr schrieb tabascoterrier <
***@***.***>:

> I've recently set up a self hosted Cloudlog instance and I'm really
> enjoying it, my thanks to everyone involved for all your hard work!
>
> Could someone tell me, what counts as a green "confirmed" QSO square on
> things like the gridsquare map?
>
> I have eSQL configured and QSOs have a green uploaded and downloaded icon.
> On the front page, my "Countries Breakdown" says "Confirmed 0 / 65 / 0",
> but all my gridsquares are red / worked but not confirmed (as are the table
> cells in things like the DXCC awards screen). So presumably there's some
> other criteria that I've yet to figure out.
>
> 73, M7SXB
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/1242>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OUNGWGIJIAZ625SFKLUHP5APANCNFSM5GGK4XDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Reply by tabascoterrier** [reply]
Thanks for the quick reply @dg9vh !

Ok, so just having a received eSQL should count as a confirmation then? Awesome, that's what I figured should be the case.

Digging around in the Gridsquares model, in [get_band_confirmed](https://github.com/magicbug/Cloudlog/blob/f0412ce5367413fe3f2d191d7b2916512eb4aab5/application/models/Gridsquares_model.php#L91) I see the clause:

```sql
AND (COL_LOTW_QSL_RCVD = "Y" OR COL_QSL_RCVD = "Y")
```

presumably that should be something like this to include the eQSL records:

```sql
AND (COL_LOTW_QSL_RCVD = "Y" OR COL_QSL_RCVD = "Y" OR COL_EQSL_QSL_RCVD = "Y")
```

So I'm wondering if eQSL came to Cloudlog as a later addition and the maps just haven't been updated to match yet? I'm happy to get my hands into the code and make a PR if so.

**Comment by dg9vh** [comment]
Hm... this is a thing that should be decided by @magicbug Peter himself :-)
I don't know on what this is based. Maybe there are some awards only
counting LoTW and traditional QSL - who knows...

So if there is this we should consider having a switching function there to
have also eqsl counted into this (if one wants them).

73 de Kim

Am Mo., 18. Okt. 2021 um 15:01 Uhr schrieb tabascoterrier <
***@***.***>:

> Thanks for the quick reply @dg9vh <https://github.com/dg9vh> !
>
> Ok, so just having a received eSQL should count as a confirmation then?
> Awesome, that's what I figured should be the case.
>
> Digging around in the Gridsquares model, in get_band_confirmed
> <https://github.com/magicbug/Cloudlog/blob/f0412ce5367413fe3f2d191d7b2916512eb4aab5/application/models/Gridsquares_model.php#L91>
> I see the clause:
>
> AND (COL_LOTW_QSL_RCVD = "Y" OR COL_QSL_RCVD = "Y")
>
> presumably that should be something like this to include the eQSL records:
>
> AND (COL_LOTW_QSL_RCVD = "Y" OR COL_QSL_RCVD = "Y" OR COL_EQSL_QSL_RCVD = "Y")
>
> So I'm wondering if eQSL came to Cloudlog as a later addition and the maps
> just haven't been updated to match yet? I'm happy to get my hands into the
> code and make a PR if so.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/magicbug/Cloudlog/discussions/1242#discussioncomment-1495426>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADKN5OSMYW4KZ5Q7BAUXO2DUHQLBZANCNFSM5GGK4XDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Amateurfunk ist das, was uns verbindet! Bleiben wir in Kontakt!
Ortsverband Q03, Neunkirchen

**Comment by AndreasK79** [comment, marked as answer]
The criteria are QSL and LoTW where it is accepted. If I remember correctly, none of the awards in Cloudlog are made to include eQSL as confirmed.

That can of course be expanded like @dg9vh says, but it is not something that will be done before V2 is released.
